firebase-web/src/main/java/io/spine/web/firebase/FirebaseNodeValue.java

```diff
@@ -0,0 +1,127 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.web.firebase;
+
+import com.google.api.client.http.ByteArrayContent;
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import com.google.firebase.database.utilities.Clock;
+import com.google.firebase.database.utilities.DefaultClock;
+import com.google.firebase.database.utilities.OffsetClock;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import static com.google.api.client.http.ByteArrayContent.fromString;
+import static com.google.common.net.MediaType.JSON_UTF_8;
+import static com.google.firebase.database.utilities.PushIdGenerator.generatePushChildName;
+
+/**
+ * The Firebase database node value.
+ */
+final class FirebaseNodeValue {
+
+    private final JsonObject value;
+
+    private FirebaseNodeValue(JsonObject value) {
+        this.value = value;
+    }
+
+    /**
+     * Creates an empty {@code FirebaseNodeValue}.
+     *
+     * <p>This is not equivalent to the {@code null} value, an empty value is supposed to be
+     * filled with entries at some point after the creation.
+     */
+    FirebaseNodeValue() {
+        this.value = new JsonObject();
+    }
+
+    /**
+     * Creates a {@code FirebaseNodeValue} whose underlying {@link com.google.gson.JsonObject} is
+     * parsed from the given {@code String}.
+     */
+    static FirebaseNodeValue from(String json) {
+        JsonParser parser = new JsonParser();
+        JsonObject value = parser.parse(json)
+                                 .getAsJsonObject();
+        return new FirebaseNodeValue(value);
+    }
+
+    /**
+     * Creates a {@code FirebaseNodeValue} which has a single entry under a generated key.
+     *
+     * @see ChildKeyGenerator
+     */
+    static FirebaseNodeValue withSingleChild(String childValue) {
+        FirebaseNodeValue nodeValue = new FirebaseNodeValue();
+        nodeValue.pushData(childValue);
+        return nodeValue;
+    }
+
+    /**
+     * Converts the value to the {@linkplain com.google.api.client.http.ByteArrayContent byte
+     * array} suitable for usage in the HTTP request.
+     */
+    ByteArrayContent toByteArray() {
+        ByteArrayContent result = fromString(JSON_UTF_8.toString(), value.toString());
+        return result;
+    }
+
+    /**
+     * Adds a child to the value.
+     *
+     * <p>The key for the child is generated via
+     * {@linkplain com.google.firebase.database.utilities.PushIdGenerator standard Firebase
+     * generation mechanism}.
+     *
+     * @return the generated key under which the data was stored
+     * @see ChildKeyGenerator
+     */
+    @CanIgnoreReturnValue
+    String pushData(String data) {
```

Comment:
I think it's more like `addChild`, similar to below. Then you'll have two overloads with almost exactly the same semantics.